app_flutter/lib/status_grid.dart

```diff
@@ -57,17 +59,18 @@ class StatusGridContainer extends StatelessWidget {
 }
 
 /// Class to handle data operations on [List<CommitStatus>].
-/// 
+///
 /// Flattens the mapping of one [CommitStatus] from many [Stage] objects,
 /// where each [Stage] object maps to many [Task] objects, to a 2D matrix.
-/// 
+///
 // TODO(chillers): Support special ordering of taskMatrix. https://github.com/flutter/cocoon/issues/478
 class StatusGridHelper {
   StatusGridHelper({@required this.statuses}) {
     _columnKeyIndex = _createTaskColumnKeyIndex(statuses);
 
-    _taskIconRow = _createTaskIconRow(statuses, _columnKeyIndex);
     _taskMatrix = _createTaskMatrix(statuses, _columnKeyIndex);
```

Comment:
You're doing a lot of different passes over the data. It would be easy to follow this algorithm if it were more compact, instead of being spread out across multiple methods.

Reply:
Do you think it would be better to add another layer of indirection and just use a `Map<int, int>` in StatusGrid that remaps the columns of the TaskMatrix to the order they should be in? Thus we never recreate the TaskMatrix unless new data comes in.

Reply:
I'm not quite sure - it's fairly difficult for me to follow along with how the task matrix is being created. One solution is to better document each step so there is a clear "1., 2. 3."

Reply:
Encapsulation could help this, create classes to encapsulate data manipulations.

For example, a TaskMatrix class encapsulates a `taskIconRow` and a list of `Columns`. It has a static factory method to create itself from a list of CommitStatus. It has a sort method to sort the columns.

A Column class encapsulates a list of `Tasks` and a `Weight`. The weight is used by TaskMatrix in sorting.

Reply:
Thank you both for the help! I was able to greatly simplify this using the feedback. I added `Column` as an encapsulation layer that made sorting much more concise, and I hope clearer to follow. I also renamed `StatusGridHelper` to `TaskMatrix`, which was discussed in the last PR, but makes much more sense to name it now.